core/cas-server-core-web/src/main/java/org/apereo/cas/config/CasCoreWebConfiguration.java

```diff
@@ -30,6 +33,24 @@
     @Autowired
     private CasConfigurationProperties casProperties;
 
+    /**
+     * Get messages from one or more property files containing values
+     * that should be exposed to Thyme templates but don't need i18n.
+     * This bean is Autowired into {@link messageSource()}.
+     * @return PropertiesFactoryBean containing all common (non-i18n) messages
+     */
+    @Bean
+    public PropertiesFactoryBean casCommonMessages() {
```

Comment:
Let me elaborate. Currently there exists this:

```java 
    @RefreshScope
    @Bean
    public HierarchicalMessageSource messageSource() {
        final CasReloadableMessageBundle bean = new CasReloadableMessageBundle();
        bean.setDefaultEncoding(casProperties.getMessageBundle().getEncoding());
        bean.setCacheSeconds(casProperties.getMessageBundle().getCacheSeconds());
        bean.setFallbackToSystemLocale(casProperties.getMessageBundle().isFallbackSystemLocale());
        bean.setUseCodeAsDefaultMessage(casProperties.getMessageBundle().isUseCodeMessage());
        bean.setBasenames(casProperties.getMessageBundle().getBaseNames());
        return bean;
    }
```

So what's happening today, is this bean is created, and then via autowiring and qualifiers it receives that common messages bean created separately. That seems inconsistent, because you should be able to internally create the same object and set it on the message source here, sort of like all the other setters above. You can still inject multiple property files into the setter and we could potentially even externalize it into a property, etc. 

Would that cause issues with FactoryBeans, etc?